examples/msaa-linelist/main.rs

```diff
@@ -0,0 +1,235 @@
+
+        let bind_group_layout =
+            device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor { bindings: &[] });
+        let bind_group = device.create_bind_group(&wgpu::BindGroupDescriptor {
+            layout: &bind_group_layout,
+            bindings: &[],
+        });
+        let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+            bind_group_layouts: &[&bind_group_layout],
+        });
+
+        let pipeline = Example::pipeline(device, &vs_module, &fs_module, &pipeline_layout, sample_count);
+        let multisampled_framebuffer = Example::multisampled_framebuffer(device, sc_desc.width, sc_desc.height, sample_count);
+
+        let mut vertex_data = vec!();
+        let mut index_data: Vec<u16> = vec!();
+
+        let max = 50;
+        for i in 0..max {
+            let percent = i as f32 / max as f32;
+            let (sin, cos) = (percent * 2.0 * std::f32::consts::PI).sin_cos();
+            vertex_data.push(Vertex {
+                _pos: [0.0, 0.0],
+                _color: [1.0, -sin, cos, 1.0],
+            });
+            vertex_data.push(Vertex {
+                _pos: [1.0 * cos, 1.0 * sin],
+                _color: [sin, -cos, 1.0, 1.0],
+            });
+
+            index_data.extend(&[
+                0 + i * 2,
+                1 + i * 2,
+            ]);
+        }
+
+        let vertex_buffer = device
+            .create_buffer_mapped(vertex_data.len(), wgpu::BufferUsage::VERTEX)
+            .fill_from_slice(&vertex_data);
+        let index_buffer = device
+            .create_buffer_mapped(index_data.len(), wgpu::BufferUsage::VERTEX)
+            .fill_from_slice(&index_data);
+
+        Example {
+            vs_module,
+            fs_module,
+            bind_group,
+            pipeline_layout,
+            pipeline,
+            multisampled_framebuffer,
+            vertex_buffer,
+            index_buffer,
+            index_count: index_data.len() as u32,
+            sample_count,
+            width: sc_desc.width,
+            height: sc_desc.height,
+        }
+    }
+
+    fn update(&mut self, event: wgpu::winit::WindowEvent, device: &mut wgpu::Device) {
+        match event {
+            wgpu::winit::WindowEvent::KeyboardInput { input, .. } => {
+                if let wgpu::winit::ElementState::Pressed = input.state {
+                    match input.virtual_keycode {
+                        Some(wgpu::winit::VirtualKeyCode::Left) => {
+                            if self.sample_count >= 2 {
+                                self.sample_count = self.sample_count >> 1;
+                                self.pipeline = Example::pipeline(device, &self.vs_module, &self.fs_module, &self.pipeline_layout, self.sample_count);
+                            }
+                        }
+                        Some(wgpu::winit::VirtualKeyCode::Right) => {
+                            if self.sample_count <= 32 {
```

Comment:
I don't think we want 64 samples supported (D3D12/Metal limitations...). Let's change this to `<32`